pkg/services/sqlstore/permissions/dashboard.go

```diff
@@ -171,17 +171,26 @@ func (f *accessControlDashboardPermissionFilter) buildClauses() {
 
 			switch f.features.IsEnabled(featuremgmt.FlagNestedFolders) {
 			case true:
-				switch f.recursiveQueriesAreSupported {
-				case true:
-					recQueryName := fmt.Sprintf("RecQry%d", len(f.recQueries))
-					f.addRecQry(recQueryName, permSelector.String(), permSelectorArgs)
+				if len(permSelectorArgs) > 0 {
+					switch f.recursiveQueriesAreSupported {
+					case true:
+						builder.WriteString("(dashboard.folder_id IN (SELECT d.id FROM dashboard as d ")
+						if len(permSelectorArgs) > 0 {
```

Comment:
nit: we already check `if len(permSelectorArgs) > 0 {` on line [174](https://github.com/grafana/grafana/pull/72620/files#diff-c4e5dd1ba75e20a5ef48ba4991af3b59ae0c29af826e5b702d9c3e5358f1a40eR174), so this check and the check on line [184](https://github.com/grafana/grafana/pull/72620/files#diff-c4e5dd1ba75e20a5ef48ba4991af3b59ae0c29af826e5b702d9c3e5358f1a40eR184) can be removed.